### PR TITLE
ERS CLI - Migration and Logs Commands

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

Changes proposed in this pull request:

- Migrate All command.
- Logs command.
- Refactoring.

**Related issue(s)**

### CODE REVIEW

File: tools/cli/go.mod [hunk 1]
```diff
@@ -19,6 +19,8 @@ require (
 	golang.org/x/net v0.0.0-20220114011407-0dd24b26b47d
 	golang.org/x/oauth2 v0.0.0-20211005180243-6b3c2da341f1
 	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.23.4
+	k8s.io/apimachinery v0.23.4
```
Comment (wozniakjan) [inline tools/cli/go.mod:23]: nit: few lines below, these get replaced
```
	k8s.io/api => k8s.io/api v0.19.12
	k8s.io/apimachinery => k8s.io/apimachinery v0.20.0			k8s.io/apimachinery => k8s.io/apimachinery v0.20.0
```
Any chance we can remove those replaces to depend on newer k8s modules?

File: tools/cli/pkg/ers/command/status.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	response, err := mothershipClient.GetClustersState(ctx, &mothership.GetClustersStateParams{
+		RuntimeID: &runtime.RuntimeID,
+	})
+	if err != nil {
+		return errors.Wrap(err, "wile getting cluster state")
+	}
+
+	defer response.Body.Close()
+
+	if IsErrResponse(response.StatusCode) {
```
Comment (wozniakjan) [inline tools/cli/pkg/ers/command/status.go:165]: nit: do we want to handle 204, 205, or 206 given this tests allows `200 <= status < 300`? I would consider making this just `response.StatusCode != http.StatusOK` if it's sufficient. Just wondering if for example ERS can send partial content with 206 and expect we fetch the rest in follow-up calls, but as currently implemented we proceed as if the content was complete.

File: tools/cli/pkg/ers/command/logs.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		panic(err.Error())
+	}
+
+	podsList, err := clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "while creatin a client set")
+	}
+
+	for _, pod := range podsList.Items {
+		if !bytes.HasPrefix([]byte(pod.Name), []byte("base-reconciler")) {
```
Comment (wozniakjan) [inline tools/cli/pkg/ers/command/logs.go:77]: I would use label selector instead of pod name prefix

**Review by wozniakjan — COMMENTED** [review]
few nits for you to consider, overall looking pretty decent

File: tools/cli/pkg/ers/client/http.go [hunk 4]
```diff
@@ -54,7 +54,7 @@ func (c *HTTPClient) get(url string) ([]ers.Instance, error) {
 		if err != nil {
 			return nil, err
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/client/http.go:55]: return nil, errors.Wrap(err, "Error while sending a GET request")

File: tools/cli/pkg/ers/command/client_utils.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+func ResponseErr(resp *http.Response) error {
+	msg, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		msg = []byte(errors.Wrap(err, "unexpected error").Error())
+	}
+	return errors.Wrapf(command.ErrMothershipResponse, "%s %d", string(msg), resp.StatusCode)
+}
+
+func StateCreatedFormatted(obj interface{}) string {
+	state := obj.(mothership.HTTPClusterStateResponse)
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/command/client_utils.go:28]: check if the assertion failed: 

```
state, ok := obj.(mothership.HTTPClusterStateResponse)
if !ok { // type assertion failed
  		// handle error
	}
```

File: tools/cli/pkg/ers/command/logs.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+type LogsCommand struct {
+	corbaCmd *cobra.Command
+	regex    string
+	log      logger.Logger
+}
+
+func (c *LogsCommand) Run() error {
+	c.log = logger.New()
+
+	fmt.Printf("%sLogs%s\n", Red, Reset)
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/command/logs.go:54]: I think "Red" is undeclared?

File: tools/cli/pkg/ers/command/logs.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/kyma-project/control-plane/tools/cli/pkg/logger"
+	"github.com/pkg/errors"
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/command/logs.go:11]: replace `"github.com/pkg/errors"` with
`"errors"` as this is now part auf standard library and `pkg/errors` was archived

File: tools/cli/pkg/ers/command/migrate_all.go [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+import (
+	"bufio"
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers/client"
+	"github.com/pkg/errors"
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/command/migrate_all.go:12]: replace "github.com/pkg/errors" with
"errors" as this is now part auf standard library and pkg/errors was archived

File: tools/cli/pkg/ers/command/status.go [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+	"regexp"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/common/runtime"
+	mothership "github.com/kyma-project/control-plane/components/reconciler/pkg"
+	mothershipAuth "github.com/kyma-project/control-plane/components/reconciler/pkg/auth"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/command"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/ers/client"
+	"github.com/kyma-project/control-plane/tools/cli/pkg/logger"
+	"github.com/pkg/errors"
```
Comment (tillknuesting) [inline tools/cli/pkg/ers/command/status.go:18]: replace "github.com/pkg/errors" with
"errors" as this is now part auf standard library and pkg/errors was archived

File: tools/cli/go.mod (hunk 1 above)
Comment (ralikio) [reply tools/cli/go.mod:23]: I was unable to make it work without `replace` but I have updated replace entries to the newest version.

File: tools/cli/pkg/ers/command/status.go (hunk 2 above)
Comment (ralikio) [reply tools/cli/pkg/ers/command/status.go:166]: This code snippet does not refer to ERS call.

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**